padertorch/contrib/examples/speaker_classification/supervised/data.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import torch
+
+from lazy_dataset.database import JsonDatabase
+
+import paderbox as pb
+import padertorch as pt
+from padertorch.ops.sequence.pack_module import pad_sequence
+from padertorch.contrib.je.data.transforms import LabelEncoder
+
+
+def get_datasets(
+    storage_dir, database_json, dataset, batch_size=16, prefetch=True
+):
+    db = JsonDatabase(database_json)
+    ds = db.get_dataset(dataset)
+
+    def prepare_example(example):
+        example['audio_path'] = example['audio_path']['observation']
+        example['speaker_id'] = example['speaker_id'].split('-')[0]
+        return example
+
+    ds = ds.map(prepare_example)
+
+    speaker_encoder = LabelEncoder(
+        label_key='speaker_id', storage_dir=storage_dir, to_array=True
+    )
+    speaker_encoder.initialize_labels(dataset=ds, verbose=True)
+    ds = ds.map(speaker_encoder)
+
+    train_set, validate_set, test_set = train_test_split(ds)
```

Comment:
Could you add a small comment to the code, why you split the train set?

Reply:
https://github.com/fgnt/padertorch/pull/57/commits/a3688c1ebb4ac61cf734e2428cd578027456742e